python/ray/tests/test_failure.py

```diff
@@ -628,6 +629,39 @@ def check_actor_restart():
     ray.get(obj2)
 
 
+def test_final_user_exception(ray_start_regular, propagate_logs, caplog):
+    class MyFinalException(Exception):
+        def __init_subclass__(cls, /, *args, **kwargs):
+            raise TypeError("Cannot subclass special typing classes")
```

Comment:
So this means if a non TypeError is raised, our code won't work

Reply:
Seems the convention is to raise TypeError: https://stackoverflow.com/questions/72639982/python-metaclass-or-init-subclass-idiomatic-exception-during-bad-subclas